cisc/certificate.go

```diff
@@ -0,0 +1,321 @@
+SjELMAkGA1UEBhMCVVMxFjAUBgNVBAoTDUxldCdzIEVuY3J5cHQxIzAhBgNVBAMT
+GkxldCdzIEVuY3J5cHQgQXV0aG9yaXR5IFgzMIIBIjANBgkqhkiG9w0BAQEFAAOC
+AQ8AMIIBCgKCAQEAnNMM8FrlLke3cl03g7NoYzDq1zUmGSXhvb418XCSL7e4S0EF
+q6meNQhY7LEqxGiHC6PjdeTm86dicbp5gWAf15Gan/PQeGdxyGkOlZHP/uaZ6WA8
+SMx+yk13EiSdRxta67nsHjcAHJyse6cF6s5K671B5TaYucv9bTyWaN8jKkKQDIZ0
+Z8h/pZq4UmEUEz9l6YKHy9v6Dlb2honzhT+Xhq+w3Brvaw2VFn3EK6BlspkENnWA
+a6xK8xuQSXgvopZPKiAlKQTGdMDQMc2PMTiVFrqoM7hD8bEfwzB/onkxEz0tNvjj
+/PIzark5McWvxI0NHWQWM6r6hCm21AvA2H3DkwIDAQABo4IBfTCCAXkwEgYDVR0T
+AQH/BAgwBgEB/wIBADAOBgNVHQ8BAf8EBAMCAYYwfwYIKwYBBQUHAQEEczBxMDIG
+CCsGAQUFBzABhiZodHRwOi8vaXNyZy50cnVzdGlkLm9jc3AuaWRlbnRydXN0LmNv
+bTA7BggrBgEFBQcwAoYvaHR0cDovL2FwcHMuaWRlbnRydXN0LmNvbS9yb290cy9k
+c3Ryb290Y2F4My5wN2MwHwYDVR0jBBgwFoAUxKexpHsscfrb4UuQdf/EFWCFiRAw
+VAYDVR0gBE0wSzAIBgZngQwBAgEwPwYLKwYBBAGC3xMBAQEwMDAuBggrBgEFBQcC
+ARYiaHR0cDovL2Nwcy5yb290LXgxLmxldHNlbmNyeXB0Lm9yZzA8BgNVHR8ENTAz
+MDGgL6AthitodHRwOi8vY3JsLmlkZW50cnVzdC5jb20vRFNUUk9PVENBWDNDUkwu
+Y3JsMB0GA1UdDgQWBBSoSmpjBH3duubRObemRWXv86jsoTANBgkqhkiG9w0BAQsF
+AAOCAQEA3TPXEfNjWDjdGBX7CVW+dla5cEilaUcne8IkCJLxWh9KEik3JHRRHGJo
+uM2VcGfl96S8TihRzZvoroed6ti6WqEBmtzw3Wodatg+VyOeph4EYpr/1wXKtx8/
+wApIvJSwtmVi4MFU5aMqrSDE6ea73Mj2tcMyo5jMd6jmeWUHK8so/joWUoHOUgwu
+X4Po1QYz+3dszkDqMp4fklxBwXRsW10KXzPMTZ+sOPAveyxindmjkW8lGy+QsRlG
+PfZ+G6Z6h7mjem0Y+iWlkYcV4PIWL1iwBi8saCbGS5jN2p8M+X+Q7UNKEkROb3N6
+KOqkqm57TH2H3eDJAkSnh6/DNFu0Qg==
+-----END CERTIFICATE-----`
+
+func compare(cOld, cNew, k string) {
+	log.Info("Compare old against new cert: " + k)
+	cert1, err := pemToCertificate(cOld)
+	if err != nil {
+		log.Info("the old value is not a cert")
+		return
+	}
+	cert2, err := pemToCertificate(cNew)
+	if err != nil {
+		log.Info("the new value is not a cert")
+		return
+	}
+	if cert1.Equal(cert2) {
+		log.Info("No change in the certificate")
+
+	} else {
+		log.Info("The new certificate is different :")
+		if !((cert1.NotAfter).Equal(cert2.NotAfter)) {
+			log.Info("    Expiration date modification: " + cert1.NotAfter.String() + " to " + cert2.NotAfter.String())
+		}
+		if !((cert1.Subject.CommonName) == (cert2.Subject.CommonName)) {
+			log.Info("    Subject modification: " + cert1.Subject.CommonName + " to " + cert2.Subject.CommonName)
+		}
+		if !((cert1.Issuer.CommonName) == (cert2.Issuer.CommonName)) {
+			log.Info("    Issuer modification: " + cert1.Issuer.CommonName + " to " + cert2.Issuer.CommonName)
+		}
+
+		if !((cert1.PublicKey.(*rsa.PublicKey).N).Cmp(cert2.PublicKey.(*rsa.PublicKey).N) == 0) {
+			log.Info("    PublicKey modification")
+		}
+	}
+}
+
+func pemToCertificate(certPem string) (*x509.Certificate, error) {
+	block, _ := pem.Decode([]byte(certPem))
+	if block == nil {
+		return nil, errors.New("Fail decode pem")
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	return cert, nil
+}
+
+func isCert(cert string) bool {
+	_, err := pemToCertificate(cert)
+	return err == nil
+}
+func check(cert string) bool {
```

Comment:
Is this `gofmt`? Usually there should be a newline between two methods.

Reply:
Or perhaps use `goimports`

Reply:
If you want to keep the `bool` for return, call this `isValid`. Else return an `error`.